site/content/entry/citra-progress-report-2018-q2.md

```diff
@@ -0,0 +1,305 @@
+at 200% speed for 2 hours. As its running at twice the max speed, playing for
+2 hours in real-time should reflect 4 hours simulated time. But, due to Citra
+sending your PC's time to games, it wasn't happening.
+
+{{< figure src="/images/entry/citra-progress-report-2018-q2/system-clock.png"
+    title="System Time setting (edited to show both options)" >}}
+
+Also, players might want to set Citra's time to a specific value to ensure
+consistency with TAS or with RNG(random number generator) or for getting
+time specific game events. Earlier you had to modify your PC's system time
+to achieve this but thanks to B3N30, you can now just set it in Citra's
+configuration window.
+
+## [Joystick Hot-plugging](https://github.com/citra-emu/citra/pull/4141) by [B3N30](https://github.com/b3n30)
+
+For a long time, Citra didn't support controller hot plugging. This meant that
+if by any chance your controller disconnected Citra would fail to detect it
+after re-connection, unless you restarted Citra. This caused a lot of
+frustration among users who preferred playing with controllers.
+
+Hot plugging (also called hot swapping) is the ability to add and remove
+devices to a computer system while the computer is running and have the
+operating system automatically recognize the change.
+
+Thanks to the efforts of B3N30 and [jroweboy](https://github.com/jroweboy)-
+who researched SDL controller configuration, Citra now doesn't fail to detect
+your controllers in case of a disconnection. However due to heavy rework of
+the existing input configuration code, this requires users to reconfigure their
+controllers after updating Citra.
+
+## UDP client to provide motion & touch controls ([#4049](https://github.com/citra-emu/citra/pull/4049), [#4059](https://github.com/citra-emu/citra/pull/4059)) by [jroweboy](https://github.com/jroweboy) and [zhaowenlan1779](https://github.com/zhaowenlan1779)
+
+Citra's previous motion controls were a bit uncomfortable to use. You had to
+use your mouse's right-click and drag to tilt the screen. But it was often
+hard to figure out where to move the cursor to get the kind of tilt we want.
+
+Using a real device for motion controls is simply much more intuitive and will
+be an all around better experience for users. So jroweboy set out to implement
+cemuhook motion/touch protocol which adds the ability for users to connect
+several different devices to citra to send direct motion and touch data to citra.
+
+{{< figure src="/images/entry/citra-progress-report-2018-q2/cemuhookudp.png"
+    title="Motion/Touch Configuration Window" >}}
+
+[Cemuhook](https://cemuhook.sshnuke.net/) is a very popular plugin developed by [rajkosto](https://github.com/rajkosto)
+for Cemu - the Wii U emulator. This protocol was chosen simply because of the
+fact that there aren't many alternatives and this already has good device
+support. Moreover, as it uses UDP, it works on just about anything with a
```

Comment:
This sounds kinda hostile to CEMU, which we shouldn't be implying here.

Reply:
which part is hostile to it ? :thinking:

Reply:
eh, ignore me.